### PR TITLE
Repair dist builds, restore binaries in main Dockerfile

### PR DESCRIPTION
**Pull request by zone117x**
* Repair dist builds -- recently broken with the new workspace configuration
* Restore binaries in main Dockerfile -- only the testnet binary was being included
* Use cargo `cfg` target dependency feature for the `sha2-asm` package so it is only included on support platforms:
  * Fixes build on Windows (msvc)
  * Fix building for non-x86_64 targets (e.g. ARM)
* Refactor the bash build scripts into more robust docker buildkit files
* Enable the `native-tls-vendored` feature for `reqwest` to fix openssl src build issues and `error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or director` error on many linux targets

Closes https://github.com/blockstack/stacks-blockchain/issues/1349 "cargo build on Windows fails"

TODO: 
* [x] Update docs with ARM fix at the cargo build level (no more command line switches should be required)
* [x] Test building on Windows with cargo
* [x] Test dist for Windows
* [x] Test dist for Alpine
* [x] Test dist for Mac
* [ ] Pull in @lgalabru's Release upload work from https://github.com/blockstack/stacks-blockchain/pull/1464/commits/f4600e7c18038ca5ffba125437b422f699f463d5
* [ ] Upload artifacts to the Stacks chain dist google cloud bucket (currently this is done manually using a script in the sidecar)

### CODE REVIEW

**Comment by zone117x** [comment]
@dantrevino Could you take a look at the build changes related to ARM?

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Deploy
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
Comment (kantai) [inline .github/workflows/deploy.yml:7]: Should this run on PRs to `master`, or just once those PRs are merged?

Comment (zone117x) [reply .github/workflows/deploy.yml:7]: I often find myself wanting to use a dist for a given PR before it's merged to master. I'm not opposed to that limitation if there's a good reason

Comment (kantai) [reply .github/workflows/deploy.yml:7]: Okay, that works for me

**Pull request merged**